### PR TITLE
Clean up the artifacts on master after archiving them elsewhere

### PR DESCRIPTION
**Pull request by adamfarley**
This to prevent artifact accumulation, and eventual issues due
to an over-filled master node.

Signed-off-by: Adam Farley <adfarley@redhat.com>

### CODE REVIEW

**Comment by sxa** [comment]
Fixes https://github.com/AdoptOpenJDK/openjdk-build/issues/2350

Questions
- This PR mentions Windows MSIs as well, but I haven't (so far) seen any indication that those are affected. Do we believe it's a problem there too?
- Removing them after completion would seem to be a change that would make it harder to debug as the "current" one will disappear. Can we perform this delete earlier in the process somewhere so tha the most recent artefacts are retained?

**Comment by adamfarley** [comment]
>     * This PR mentions Windows MSIs as well, but I haven't (so far) seen any indication that those are affected. Do we believe it's a problem there too?

No, that was me covering all bases. The change has been modified to only affect Mac packages.

>     * Removing them after completion would seem to be a change that would make it harder to debug as the "current" one will disappear. Can we perform this delete earlier in the process somewhere so tha the most recent artefacts are retained? Perhaps before the copy from the installer job?

Good idea. Change made.

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (sxa) [inline pipelines/build/common/openjdk_build_pipeline.groovy:511]:
```suggestion
    For Mac, we also clean up pkgs on master node from previous runs, if needed (Ref openjdk-build#2350
```

**Review by sxa — APPROVED** [review]
Suggested change to include a reference to the issue that explains why this change is required, but otherwise LGTM. Hopefully we can run a quick pipeline without tests to verify once this is merged.

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (adamfarley) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: Done.

Comment (adamfarley) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: Will remember to do this in future.

**Review by M-Davies — APPROVED** [review]
LGTM

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:522]: Shouldn't this be in a cleanup function of somesort?  I'm curious as to why we don't explicitly do this for say, Windows.

Comment (adamfarley) [reply pipelines/build/common/openjdk_build_pipeline.groovy:522]: I'm honestly uncertain as to why we don't need to do this for Windows. The msi files don't seem to accumulate like the pkg files do.

I don't think this needs an entire function. I think the function should be consistent in how much trash it leaves behind, and since we don't create trash from a Linux run, we shouldn't do for a Mac run either.

That said, I see no reason we couldn't have a "cleanWs" at the start/end of a run. Perhaps an enhancement?

Comment (karianna) [reply pipelines/build/common/openjdk_build_pipeline.groovy:522]: Yeah we can open a separate issue for that I guess.

**Comment by adamfarley** [comment]
Ok, the checks have passed, but we should wait until tomorrow morning for the merge so we can give it a run without risking the nightlies.

**Comment by adamfarley** [comment]
Doesn't look like any of the mac nightlies are queued, so let's get this merged and a test run done.

**Comment by adamfarley** [comment]
Post-merge testing seems to be blocked on a networking issue.

https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk15u/job/jdk15u-mac-x64-hotspot/165/console
11:01:49  * What went wrong:
11:01:49  Could not determine the dependencies of task ':shadowJar'.
11:01:49  > Could not resolve all dependencies for configuration ':runtimeClasspath'.
11:01:49     > Could not resolve org.jenkins-ci.main:jenkins-core:2.85.
11:01:49       Required by:
11:01:49           project :
11:01:49        > Could not resolve org.jenkins-ci.main:jenkins-core:2.85.
11:01:49           > Could not get resource 'https://repository.mulesoft.org/nexus/content/repositories/public/org/jenkins-ci/main/jenkins-core/2.85/jenkins-core-2.85.pom'.
11:01:49              > Could not GET 'https://repository.mulesoft.org/nexus/content/repositories/public/org/jenkins-ci/main/jenkins-core/2.85/jenkins-core-2.85.pom'.
11:01:49                 > Read timed out
11:01:49     > Could not resolve org.jenkins-ci.plugins.workflow:workflow-step-api:2.22.
11:01:49       Required by:
11:01:49           project :
11:01:49        > Could not resolve org.jenkins-ci.plugins.workflow:workflow-step-api:2.22.
11:01:49           > Could not get resource 'https://repository.mulesoft.org/nexus/content/repositories/public/org/jenkins-ci/plugins/workflow/workflow-step-api/2.22/workflow-step-api-2.22.pom'.
11:01:49              > Could not GET 'https://repository.mulesoft.org/nexus/content/repositories/public/org/jenkins-ci/plugins/workflow/workflow-step-api/2.22/workflow-step-api-2.22.pom'.
11:01:49                 > Read timed out

**Comment by sxa** [comment]
Above error does not appear to have stopped the build progressing

**Comment by adamfarley** [comment]
Stewart is right. It prevented the last build from progressing (164), so I made assumptions when 165 saw the same error. The file we timeout trying to access now seems accessible though (and it wasn't earlier), so I'm putting 164 down as "upstream server glitch" and moving on.

[Here's an extra PR to remove the json and sha256 files as well.](https://github.com/AdoptOpenJDK/openjdk-build/pull/2363)

**Comment by adamfarley** [comment]
The only pkg files left [(in this build run)](https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk15u/job/jdk15u-mac-x64-hotspot/165/console) are the ones from this build. Success!

```
11:41:07  + sha256sum workspace/target/OpenJDK15U-jdk_x64_mac_hotspot_2021-01-07-10-59.pkg

11:41:09  + sha256sum workspace/target/OpenJDK15U-jre_x64_mac_hotspot_2021-01-07-10-59.pkg
```

**Pull request merged**
